modules/gdnative/videodecoder/video_stream_gdnative.cpp

```diff
@@ -125,7 +125,12 @@ bool VideoStreamPlaybackGDNative::open_file(const String &p_file) {
 		texture_size = *(Vector2 *)&vec;
 
 		pcm = (float *)memalloc(num_channels * AUX_BUFFER_SIZE * sizeof(float));
-		memset(pcm, 0, num_channels * AUX_BUFFER_SIZE * sizeof(float));
+
+		//VideoStreamGDNativeAudioFix Branch
+		//Only do memset if num_channels>0 otherwise it will be crash
```

Comment:
Better than only protecting the line that crashes, it would be better to handle the actual logic of what should or should not be done when `num_channels` in 0. For example the `pcm` assignment line is also irrelevant as it's running `memalloc(0)` - yet `pcm` is a member variable so it likely still needs to be given a value that identifies it as invalid. Is 0 correct, or might it keep errors that are not handled here?

CC @kidrigger

Reply:
I think a better way to go about it is to keep a boolean "has_audio" variable.
So in case of lack of audio - all the audio code can be skipped.

Reply:
Hi @akien-mga , @kidrigger sorry for the delay. @akien-mga you are right i will fix it. @kidrigger your idea could be also a good solition but I think that num_channels it is enough to notice about if video file has audio or not. In fact i use num_channels as you propose but in this case not a bool variable. Any way thanks for the suggestion.